runtime-decompiler/src/main/java/org/jrd/backend/data/ArchiveManager.java

```diff
@@ -0,0 +1,58 @@
+package org.jrd.backend.data;
+
+import org.jrd.backend.communication.FsAgent;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ArchiveManager {
+    File c;
+    String tmpdir = System.getProperty("java.ioo.tmpdir");
+    ArrayList<String> currentJars = new ArrayList<>();
+
+    public ArchiveManager(File c) {
+        this.c = c;
+    }
+
+    public boolean isClassInFile(String clazz) throws IOException {
+        ZipInputStream zis = new ZipInputStream(new FileInputStream(c));
+        if (!currentJars.isEmpty()) {
+            currentJars = new ArrayList<>();
+        }
+        currentJars.add(c.getName());
+        var ret = findClazz(zis, clazz);
+        if (!ret) {
+            currentJars.remove(c.getName());
+        }
+        return ret;
+    }
+
+    private boolean findClazz(ZipInputStream zis, String clazz) throws IOException {
+        ZipEntry entry = null;
+        while ((entry = zis.getNextEntry()) != null) {
+            if (!entry.isDirectory()) {
+                if (entry.getName().endsWith(".jar") || entry.getName().endsWith(".zip")) {
```

Comment:
JAR or ZIP or war or ear or module or whatever... Also, inst that used more times in the code?
I woudl recomedn to create some method which tells us what to do, returning DIR xor ZIP xor CLASS xor RESOURCE
Where dir and .class is strigthforward. Everythig else zip or some damn resource. Zip you can find by suffix, but is it enough? Considering the ncountless number of suffixes? So maybe try to open it as zip (once you checkknown suffixes) andif it opens, then it is zip, otherwise resource.  I'm not sure how expensive the opening as zip may be. So it may enforce some caching of the indirect types.

Reply:
It would make sense to extract it into a function (probably static, as it is also used in FsAgent) and would not change at all.
How to rewrite, what all to check... Probably will deal with as a last thing.

Reply:
tbh, I'm not sure how toactually write a proper nested in nested jar "url"

Reply:
But maybe we need that mapper? I was surprised it is nowhere in the PR. SOme structureholding reocrd from what (nested) file it arrived.

Reply:
But most likley that wil lpaly to effect only with editing enabled.